airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java

```diff
@@ -196,21 +196,19 @@ private MediaFile checkLastModified(MediaFile mediaFile, boolean useFastCache) {
         List<MediaFile> result = new ArrayList<MediaFile>();
         for (MediaFile child : mediaFileDao.getChildrenOf(parent.getPath())) {
             child = checkLastModified(child, useFastCache);
-            if (child.isDirectory() && includeDirectories && includeMediaFile(child)) {
-                result.add(child);
-            }
-            if (child.isFile() && includeFiles && includeMediaFile(child)) {
-                result.add(child);
+            if (includeMediaFile(child)) {
+                if (child.isDirectory() && includeDirectories) {
+                    result.add(child);
+                }
+                if (child.isFile() && includeFiles) {
+                    result.add(child);
+                }
             }
         }
 
         if (sort) {
             Comparator<MediaFile> comparator = new MediaFileComparator(settingsService.isSortAlbumsByYear());
-            // Note: Intentionally not using Collections.sort() since it can be problematic on Java 7.
-            // http://www.oracle.com/technetwork/java/javase/compatibility-417013.html#jdk7
-            Set<MediaFile> set = new TreeSet<MediaFile>(comparator);
-            set.addAll(result);
-            result = new ArrayList<MediaFile>(set);
+            result.sort(comparator);
```

Comment:
Might still be worth leaving a comment here about Java7 concern even if we no longer support it.

Reply:
This pattern is used in a lot of other places, I don't think it's worth keeping the comment, especially since the public version of Java7 isn't supported [since 2015]( https://en.wikipedia.org/wiki/Java_version_history ).